ArduPlane/quadplane.cpp

```diff
@@ -2577,8 +2659,20 @@ bool QuadPlane::verify_vtol_land(void)
     if (!available()) {
         return true;
     }
-    if (poscontrol.state == QPOS_POSITION2 &&
-        plane.auto_state.wp_distance < 2) {
+
+#if PRECISION_LANDING == ENABLED
+    AC_PrecLand &precland = plane.precland;
+    const bool navigating = pos_control->is_active_xy();
+    bool doing_precision_landing = precland.target_acquired() && navigating;
+    Vector2f target;
+    plane.precland.get_target_position_relative_cm(target);
+#else
+    bool doing_precision_landing = false;
+#endif
+
+    if (poscontrol.state == QPOS_POSITION2 && 
+        ((plane.auto_state.wp_distance < 2) || (doing_precision_landing && target.length() < 2))) {
```

Comment:
I think we could abstract this into a function that gets the XY distance to target